src/java/org/apache/cassandra/index/sai/disk/v1/segment/SegmentBuilder.java

```diff
@@ -71,10 +73,48 @@
     private ByteBuffer minTerm;
     private ByteBuffer maxTerm;
 
+    final AbstractType<?> termComparator;
     long totalBytesAllocated;
     int rowCount = 0;
     int maxSegmentRowId = -1;
 
+    public static class KDTreeSegmentBuilder extends SegmentBuilder
+    {
+        protected final byte[] buffer;
+        private final BKDTreeRamBuffer kdTreeRamBuffer;
+
+        public KDTreeSegmentBuilder(AbstractType<?> termComparator, NamedMemoryLimiter limiter)
+        {
+            super(termComparator, limiter);
+
+            int typeSize = TypeUtil.fixedSizeOf(termComparator);
+            this.kdTreeRamBuffer = new BKDTreeRamBuffer(typeSize);
+            this.buffer = new byte[typeSize];
+            this.totalBytesAllocated = this.kdTreeRamBuffer.ramBytesUsed();
```

Comment:
Nit: we don't need the `this.`:
```suggestion
            kdTreeRamBuffer = new BKDTreeRamBuffer(typeSize);
            buffer = new byte[typeSize];
            totalBytesAllocated = kdTreeRamBuffer.ramBytesUsed();
```